samples: matter: update manifest and disable Read Client (#13464)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

------

* Updated the page about :ref:`ug_matter_device_low_power_configuration` with the information about Intermittently Connected Devices (ICD) configuration.

> * Added kconfig for disabling and enabling Read Client functionality.

Reviewer: ```suggestion
* Added a Kconfig option for disabling or enabling Read Client functionality.
```

> * Added a Kconfig option for disabling or enabling :ref:`Read Client functionality <ug_matter_configuring_read_client>`.

Reviewer: ```suggestion
* Added a Kconfig option for disabling or enabling :ref:`ug_matter_configuring_read_client`.
```

  * Fixed an issue that prevented nRF Toolbox for iOS in version 5.0.9 from controlling the sample using :ref:`nus_service_readme`.
  * Changed the design of the :ref:`matter_lock_sample_wifi_thread_switching` feature so that support for both Matter over Thread and Matter over Wi-Fi is included in a single firmware image.

> * Disabled Read Client functionality in all Matter samples except :ref:`matter_light_switch_sample`, and :ref:`matter_bridge_app`.

Reviewer: ```suggestion
* Disabled Read Client functionality in all Matter samples except the :ref:`matter_light_switch_sample` sample, and the :ref:`matter_bridge_app` application.
```

> * Disabled :ref:`ug_matter_configuring_read_client` in most Matter samples using the new :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` Kconfig option.

Reviewer: Just realized this should come before the Door Lock sample information and after the unification of buttons and LEDs that is not present in this PR. I think this will require a rebase to avoid merge conflicts.

Author: Moved, and rebased :)

> * Disabled :ref:`Read Client functionality <ug_matter_configuring_read_client>` in most Matter samples using the new :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` Kconfig option.

Reviewer: ```suggestion
* Disabled :ref:`ug_matter_configuring_read_client` in most Matter samples using the new :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` Kconfig option.
```

> * Disabled Read Client functionality in all Matter samples except the :ref:`matter_light_switch_sample` and :ref:`matter_thermostat_sample` samples, and the :ref:`matter_bridge_app` application.

Reviewer 2: Minor: I'm not sure about it, but has it ever been enabled? I mean, this is an configuration option that was likely not present in Matter 1.1 and NCS 2.5.0. So informing the user that something that was not enabled before was disabled may not be useful.

Reviewer: I'm of two minds. The entry in the protocol section makes it sound like the functionality was always there, but the configuration option to disable it was added. In fact, let me make a new suggestion that incorporates the Kconfig option in the entry to add some clarity.

> * Disabled Read Client functionality in all Matter samples except the :ref:`matter_light_switch_sample` and :ref:`matter_thermostat_sample` samples, and the :ref:`matter_bridge_app` application.

Reviewer: ```suggestion
* Disabled Read Client functionality in most Matter samples using the new :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` Kconfig option.
  The functionality is not disabled in the :ref:`matter_light_switch_sample` and :ref:`matter_thermostat_sample` samples, or the :ref:`matter_bridge_app` application.
```
